### DISCUSSION TITLE
Follow repository renaming when resolving actions to avoid breaking old references?

### DISCUSSION THREAD

**Discussion by mkniewallner** [community/community#123611] [Actions]
### Select Topic Area

Product Feedback

### Body

Recently, [googleapis/release-please-action](https://github.com/googleapis/release-please-action) repository hosting an action used in quite a lot of repositories was moved from `google-github-actions` organisation to `googleapis` (see https://github.com/googleapis/release-please-action/issues/980).

When a GitHub repository is renamed, or moved from one organisation to another, it is still possible to perform a checkout using the old reference. For instance in the case above, it's still possible to do:
```shell
git clone git@github.com:google-github-actions/release-please-action.git
```

So there is a retro-compatibility mechanism in place to avoid breaking usages using the old reference. Unfortunately, it looks like GitHub Actions do not benefit from this mechanism, as when using:
```yaml
jobs:
  a-job:
    runs-on: ubuntu-22.04
    steps:
      - uses: google-github-actions/release-please-action@v4
```
we end up having a resolution error:
> Error: Unable to resolve action. Repository not found: google-github-actions/release-please-action

Is this intended? Should Actions resolution follow the same mechanism, and still work if you use old references? Otherwise, it seems that it's almost impossible to move or rename a repository without breaking usages of the action it hosts.

_Note that this resolution error happens regardless of the pattern used for specifying the reference to the action (`latest`, specific major, specific commit SHA1, ...)._

**Comment by Th3S4mur41** [comment]
Same issue happened to us when we had to rename an organization.
Redirects worked fine for checkout and for any kind of links pointing to the old name.
Even the marketplace entries were following the redirects.

Only actions were not (reliably) as reported e.g. in https://github.com/atos-actions/clean-self-hosted-runner/issues/290
